Towny/src/main/java/com/palmergames/bukkit/towny/db/TownyFlatFileSource.java

```diff
@@ -465,11 +466,17 @@ public boolean loadResident(Resident resident) {
 								universe.unregisterResident(olderRes);
 							} catch (NotRegisteredException ignored) {}
 							// Check if the older resident is a part of a town
-							if (olderRes.hasTown()) {
+							Town olderResTown = olderRes.getTownOrNull();
+							if (olderResTown != null) {
 								try {
 									// Resident#removeTown saves the resident, so we can't use it.
-									olderRes.getTown().removeResident(olderRes);
-								} catch (NotRegisteredException ignored) {}
+									olderResTown.removeResident(olderRes);
+								} catch (EmptyTownException e) {
+									try {
+										universe.unregisterTown(olderResTown);
+									} catch (NotRegisteredException ignored) {}
+									deleteTown(olderResTown);
```

Comment:
```suggestion
									TownyUniverse.getInstance().getDataSource().removeTown(town, DeleteTownEvent.Cause.NO_RESIDENTS, null, false);
```

Reply:
Alternatively we could just call resident.removeTown() and this would be taken care of I think.

Reply:
The town & townblocks are not loaded yet at this point so I figured we could get away with unregistering it and deleting the file

Reply:
touché